code/game/objects/items/weapons/cosmetics.dm

```diff
@@ -491,3 +491,71 @@
 		return
 	if (prob(25))
 		shatter()
+
+/obj/item/weapon/nanitecontacts
+	name = "Nanite Contacts"
+	desc = "Deploys nanobots to your eyes to change their color."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "nanite_contact"
+	flags = FPRINT
+	w_class = W_CLASS_TINY
+	var/color_r = 255
+	var/color_g = 255
+	var/color_b = 255
+
+/obj/item/weapon/nanitecontacts/New()
+	..()
+	color_r = rand(0,255)
+	color_g = rand(0,255)
+	color_b = rand(0,255)
+	update_icon()
+
+/obj/item/weapon/nanitecontacts/update_icon()
+	overlays.len = 0
+	var/icon/contact_color = new/icon("icon" = 'icons/obj/items.dmi', "icon_state" = "contacts_overlay")
+	contact_color.Blend(rgb(color_r, color_g, color_b), ICON_ADD)
+	overlays += contact_color
+
+/obj/item/weapon/nanitecontacts/attack_self(mob/user as mob)
+	var/new_color = input(user, "Choose the contact's color:", "Color Select") as color|null
+	if(new_color)
```

Comment:
This should check whether the user is still alive and still has the contacts in their hands.